### PR TITLE
Update periodic and bulk payments models and descriptions

### LINKED ISSUE #33: Remove or complete equivalentAmount
In the table which describes all properties for each payment type there is a property `equivalentAmount` which is missing in the model `paymentInitiation_json`.

Either we remove that property from the description or we add it to the model.

**NOTE: There is a corresponding table in periodic payments and bulk payments. Please apply the above corrections to ALL 3 tables and footnotes.**

### LINKED ISSUE #35: Missing property `debtorAgent` in periodic payments
In the table which describes all properties for each payment type there is a property `debtorAgent` which is missing in the model `periodicPaymentInitiation_json` but is existing in the model `paymentInitiation_json`.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #36: Missing property `debtorName` in periodic payments
In the table which describes all properties for each payment type there is a property `debtorName` which is missing in the model `periodicPaymentInitiation_json` but is existing in the model `paymentInitiation_json`.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #37: Missing property `debtorId` in periodic payments
In the table which describes all properties for each payment type there is a property `debtorId` which is missing in the model `periodicPaymentInitiation_json` but is existing in the model `paymentInitiation_json`.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #38: Missing properties `equivalentAmount` and `currencyOfTransfer` in periodic payment
In the table which describes all properties for each payment type there are properties `currencyOfTransfer` and `equivalentAmount` which is missing in the model periodicPaymentInitiation_json but is existing in the model paymentInitiation_json.

Either we remove that property from the description or we add it to the model.

**Note: In the issue #31 we discuss whether to use `transactionCurrency` or `currencyOfTransfer`. Please use the same name as in that issue.**

### LINKED ISSUE #39: Missing property `creditorId` in periodic payments
In the table which describes all properties for each payment type there is a property `creditorId` which is missing in the model periodicPaymentInitiation_json but is existing in the model paymentInitiation_json.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #40: Missing property `creditorNameAndAddress` in periodic payments
In the table which describes all properties for each payment type there is a property `creditorNameAndAddress` which is missing in the model periodicPaymentInitiation_json but is existing in the model paymentInitiation_json.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #41: Missing property `chargeBearer` in periodic payments
In the table which describes all properties for each payment type there is a property `chargeBearer` which is missing in the model periodicPaymentInitiation_json but exists in the model paymentInitiation_json.

Either we remove that property from the description or we add it to the model.

### LINKED ISSUE #42: Missing property `serviceLevel` in periodic payments
In the table which describes all properties for each payment type there is a property `serviceLevel` which is missing in the model periodicPaymentInitiation_json but exists in the model paymentInitiation_json.

Either we remove that property from the description or we add it to the model.

### PR DESCRIPTION
**Pull request by wishopt**
Update `paymentInitiationBulkElement_json` and `periodicPaymentInitiation_json` to match the changes made to `paymentInitiation_json` in #47

Adresses issues #33 #35 #36 #37 #38 #39 #40 #41 #42

### CODE REVIEW

File: swiss-ng-api.yaml [hunk 1]
```diff
@@ -6137,10 +6138,12 @@ components:
...
-#        debtorName: # N.A. for all, but can be used by ASPSP if needed
-#          $ref: "#/components/schemas/debtorName"
-#        debtorId: # N.A. for all, but can be used by ASPSP if needed
-#          $ref: "#/components/schemas/debtorId"
+        debtorName: # N.A. for all, but can be used by ASPSP if needed
+          $ref: "#/components/schemas/debtorName"
+        debtorId: # N.A. for all, but can be used by ASPSP if needed
+          $ref: "#/components/schemas/debtorId"
+        debtorAgent:
+          $ref: "#/components/schemas/deptorAgent7-CH"
```
Comment (yves-bussard) [inline swiss-ng-api.yaml]: Just so I understand this correctly:  All single payments in a bulk payment are paid from the same account (`debtorAccout`). But each payment in a bulk payment can specify a different `debtorName` (e.g. _Hans Huber_, _H. Huber_ und _Mickey Mouse_) although the `debtorAccount` is the same? I can understand that the display name may vary per payment. But what about `debtorId`? Is it really possible to have different 'debtorId' although the `debtorAccount` is the same? Same problem with the `debtorAgent`

File: swiss-ng-api.yaml [hunk 2]
```diff
@@ -6279,42 +6281,72 @@ components:
         - debtorName
         - creditorAccount
         - creditorName
+        - requestedExecutionDate
```
Comment (yves-bussard) [inline swiss-ng-api.yaml]: Isn't this in periodic payments? There should not be a `requestedExecutionDate` in periodic payment.

File: swiss-ng-api.yaml [hunk 3]
```diff
@@ -6236,12 +6236,14 @@ components:
...
-        <tr><td>remittanceInformationUnstructuredArray</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td></tr>
         <tr><td>serviceLevel</td><td>n.a.</td><td>optional</td><td>optional</td><td>optional</td><td>mandatory</td><td>optional</td></tr>
         <tr><td>remittanceInformationUnstructured</td><td>n.a.</td><td>optional</td><td>n.a.</td><td>optional</td><td>optional</td><td>optional</td></tr>
+        <tr><td>remittanceInformationUnstructuredArray</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td></tr>
         <tr><td>remittanceInformationStructured</td><td>mandatory</td><td>optional</td><td>mandatory 6)</td><td>optional</td><td>optional</td><td>optional</td></tr>
-        <tr><td>requestedExecutionDate</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td></tr>
+        <tr><td>remittanceInformationStructuredArray</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td><td>n.a.</td></tr>
+        <tr><td>requestedExecutionDate</td><td>mandatory</td><td>mandatory</td><td>mandatory</td><td>mandatory</td><td>mandatory</td><td>mandatory</td></tr>
```
Comment (yves-bussard) [inline swiss-ng-api.yaml]: Should always be n.a.

**Review by yves-bussard — CHANGES_REQUESTED** [review]

File: swiss-ng-api.yaml (hunk 1 above)
Comment (blehers) [reply swiss-ng-api.yaml]: I can understand your reasoning, we are following the Berlin Group definitions accurately here (see screenshot).
However, there are two things to consider. 

1. `debtorName` and `debtorID` are labeld "n.a." in Berlin Group's standard specification. My suggestion: no deviation from the standard and therefore no change
2. `debtorAgent` is a Swiss specific data element. My suggestion: handling it like `debtorAccount`

grafik

Comment (yves-bussard) [reply swiss-ng-api.yaml]: > `debtorAgent` is a Swiss specific data element. My suggestion: handling it like `debtorAccount`

You mean, place it on the level of the `debtorAccount` and not in `payments`?

Comment (blehers) [reply swiss-ng-api.yaml]: Yes.

File: swiss-ng-api.yaml (hunk 3 above)
Comment (blehers) [reply swiss-ng-api.yaml]: You are right, `requestedExecutionDate `has to be `n.a.`.

**Review by yves-bussard — APPROVED** [review]

**Pull request merged**